### PR TITLE
split receiver/otlpreceiver into its own module

### PR DESCRIPTION
**Pull request by codeboten**
This splits the OTLP receiver into its own module. Currently this leaves `scraperhelper` and `scrapererror` inside the main collector module, this is similar to `exporterhelper`. Note that doing this split brought up the interesting issue that the OTLP HTTP exporter depends on the OTLP receiver for some of its tests. I can address this separately from this PR.

Fixes https://github.com/open-telemetry/opentelemetry-collector/issues/6190

### CODE REVIEW

**Comment by bogdandrutu** [comment]
> Note that doing this split brought up the interesting issue that the OTLP HTTP exporter depends on the OTLP receiver for some of its tests. I can address this separately from this PR.

Why is that an issue?

**Comment by codeboten** [comment]
> Why is that an issue?

It's creating a dependency on the package just for the tests, it's better not to have a dependency between the exporter and the receiver just for tests.

**Comment by bogdandrutu** [comment]
> It's creating a dependency on the package just for the tests, it's better not to have a dependency between the exporter and the receiver just for tests.

How can you have an e2e test otherwise?

File: .chloggen/split-receiver.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext: |-
+  The import path for the OTLP receiver can now be access directly:
+  - `go.opentelemetry.io/collector/receiver/otlpreceiver`
+  
+  If using this receiver, modify your Collector builder configuration to use `gomod` directly, such as:
+  - `gomod: go.opentelemetry.io/collector/receiver/otlpreceiver v0.64.0`
```
Comment (bogdandrutu) [inline .chloggen/split-receiver.yaml:21]: new line?

File: cmd/builder/README.md [hunk 2]
```diff
@@ -10,12 +10,10 @@ $ GO111MODULE=on go install go.opentelemetry.io/collector/cmd/builder@latest
...
 exporters:
   - gomod: "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/alibabacloudlogserviceexporter v0.40.0"
-  - import: go.opentelemetry.io/collector/exporter/loggingexporter
-    gomod: go.opentelemetry.io/collector v0.40.0
+  - gomod: go.opentelemetry.io/collector/exporter/loggingexporter v0.63.0
 
 receivers:
-  - import: go.opentelemetry.io/collector/receiver/otlpreceiver
-    gomod: go.opentelemetry.io/collector v0.40.0
+  - gomod: go.opentelemetry.io/collector/receiver/otlpreceiver v0.64.0
```
Comment (bogdandrutu) [inline cmd/builder/README.md:16]: Update all versions here?

File: cmd/builder/internal/config/default.yaml [hunk 3]
```diff
@@ -6,8 +6,7 @@ dist:
   otelcol_version: 0.63.0
 
 receivers:
-  - import: go.opentelemetry.io/collector/receiver/otlpreceiver
-    gomod: go.opentelemetry.io/collector v0.63.0
+  - gomod: go.opentelemetry.io/collector/receiver/otlpreceiver v0.64.0
```
Comment (bogdandrutu) [inline cmd/builder/internal/config/default.yaml:9]: why not 63?

**Review by bogdandrutu — APPROVED** [review]

**Comment by codeboten** [comment]
> How can you have an e2e test otherwise?

The OTLP gRPC exporter uses a mock receiver, it's unclear why the OTLP HTTP exporter differs from this.

https://github.com/open-telemetry/opentelemetry-collector/blob/07603a09f7c0a37b60ec2327d3c0d4a0322b3de6/exporter/otlpexporter/otlp_test.go#L50-L74

File: cmd/builder/README.md (hunk 2 above)
Comment (codeboten) [reply cmd/builder/README.md]: Updated all versions except batch processor, that's coming in a follow up PR

File: .chloggen/split-receiver.yaml (hunk 1 above)
Comment (codeboten) [reply .chloggen/split-receiver.yaml]: Done

File: cmd/builder/internal/config/default.yaml (hunk 3 above)
Comment (codeboten) [reply cmd/builder/internal/config/default.yaml]: Fixed

**Comment by bogdandrutu** [comment]
> The OTLP gRPC exporter uses a mock receiver, it's unclear why the OTLP HTTP exporter differs from this.

That is not e2e :)

**Pull request merged**